src/optimizer/query_rewrite.c

```diff
@@ -587,22 +587,30 @@ qo_convert_attref_to_dotexpr (PARSER_CONTEXT * parser, PT_NODE * node, void *arg
 	case PT_NORMAL:
 	  /* we must transform this NAME node into a DOT node in place to preserve its address. (Otherwise, we have to
 	   * find all the places that point to it and change them all.) */
-	  arg2 = parser_copy_tree (parser, node);
-	  if (arg2)
-	    arg2->next = NULL;
-	  arg1 = pt_name (parser, attr->info.name.original);
-	  if (arg1)
-	    {
-	      arg1->info.name.resolved = rvar->info.name.original;
-	      arg1->info.name.spec_id = new_spec->info.spec.id;
-	      arg1->info.name.meta_class = PT_NORMAL;
-	      arg1->type_enum = attr->type_enum;
-	      arg1->data_type = parser_copy_tree (parser, attr->data_type);
-	    }
-	  node->node_type = PT_DOT_;
-	  node->info.dot.arg1 = arg1;
-	  node->info.dot.arg2 = arg2;
-	  node->info.dot.selector = NULL;
+	  {
+	    arg2 = parser_copy_tree (parser, node);
+	    if (arg2)
+	      {
+		arg2->next = NULL;
+	      }
+	    arg1 = pt_name (parser, attr->info.name.original);
+	    if (arg1)
+	      {
+		arg1->info.name.resolved = rvar->info.name.original;
+		arg1->info.name.spec_id = new_spec->info.spec.id;
+		arg1->info.name.meta_class = PT_NORMAL;
+		arg1->type_enum = attr->type_enum;
+		arg1->data_type = parser_copy_tree (parser, attr->data_type);
+	      }
+
+	    int coll_modifier = node->info.name.coll_modifier;
+
+	    node->node_type = PT_DOT_;
```

Comment:
I prefer to use init_info_node of PT_DOT_ to avoid possible future issues of uninitialized new added members.